src/empty.js

```diff
@@ -20,15 +25,20 @@ var _curry1 = require('./internal/_curry1');
  *      R.empty({x: 1, y: 2});  //=> {}
  */
 module.exports = _curry1(function empty(x) {
-  if (x != null && typeof x.empty === 'function') {
-    return x.empty();
-  } else if (x != null && typeof x.constructor != null && typeof x.constructor.empty === 'function') {
-    return x.constructor.empty();
-  } else {
-    switch (Object.prototype.toString.call(x)) {
-      case '[object Array]':  return [];
-      case '[object Object]': return {};
-      case '[object String]': return '';
-    }
-  }
+  return (
+    (x != null && typeof x.empty === 'function') ?
+      x.empty() :
+    (x != null && typeof x.constructor != null && typeof x.constructor.empty === 'function') ?
+      x.constructor.empty() :
+    _isArray(x) ?
+      [] :
+    _isString(x) ?
+      '' :
+    _isObject(x) ?
+      {} :
+    _isArguments(x) ?
+      (function() { return arguments; }()) :
+    // else
+      void 0
+  );
 });
```

Comment:
I'm still mixed about these deeply nested conditional operators.  That's a lot of `? - :` pairs!

I'm not saying I dislike it.  In fact, I think I could grow to like it.  But I'm not used to doing more than two levels, and that's generally in a one-liner for a comparator:

    return a < b ?  -1 : a > b : 1 : 0;

It's certainly nice to deal with simple expressions here and not control-flow statements like `if-else` or `switch`.  Hmm.